File: evidence/generated-systems/golang/inventory-on-golang/inventory-on-golang/internal/model/Quarantine.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Quarantine Declaration
//==============================================================
type Quarantine struct {
    gorm.Model
     Reason                                    string
    StartedAt                                                            time.Time
    ReleasedAt                                                            time.Time
    WarehouseId         *uint
    Warehouse           *Warehouse `gorm:"foreignKey:WarehouseId"`
     Items           []InventoryItem `gorm:"foreignKey:ItemsFromQuarantineId"`
    LotId         *uint
    Lot           *Lot `gorm:"foreignKey:LotId"`
     SerialNumbers           []SerialNumber `gorm:"foreignKey:SerialNumbersFromQuarantineId"`
    Disposition                      Disposition

// parent associations as their child

}

File: evidence/generated-systems/angular/inventory-on-angular/src/app/models/Quarantine.js

// Define collection and schema for Quarantine
export interface Quarantine {
    reason:
	type : string
    startedAt:
	type : Date
    releasedAt:
	type : Date
    Warehouse:
	type : Schema.Types.ObjectId
    Items:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InventoryItem' }]
    Lot:
	type : Schema.Types.ObjectId
    SerialNumbers:
 	type : [{ type: Schema.Types.ObjectId, ref: 'SerialNumber' }]
    Disposition:
 	type : String
#
    collection: 'quarantines'
}
